File: pset2/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <ctype.h>

int main(int argc, string argv[])
{
    int key_index = 1;
    if (argc != 2)
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }
    int key = atoi(argv[key_index]);
    printf("Key: %i\n", key);
    string plaintext = get_string("plaintext: ");
    printf("ciphertext: ");
    for (int i = 0, n = strlen(plaintext); i < n; i++)
    {
        if (islower(plaintext[i]))
        {

            printf("%c", ((plaintext[i] - 'a') + key) % 26 + 'a');
        }
        else if (isupper(plaintext[i]))
        {
            printf("%c", ((plaintext[i] - 'A') + key) % 26 + 'A');
        }
        else
        {
            printf("%c", plaintext[i]);
        }
    }
    printf("\n");
}

File: pset6/caesar.py
from cs50 import get_string
from sys import argv


def main():
    if len(argv) != 2:
        print("Usage python caesar.py key")
        exit(1)
    key = int(argv[1])
    plaintext = get_string("plaintext: ")
    ciphertext = ""
    for p in plaintext:
        if p.isalpha():
            if p.isupper():
                ciphertext += chr(((ord(p) - ord('A')) + key) % 26 + ord('A'))
            elif p.islower():
                ciphertext += chr(((ord(p) - ord('a')) + key) % 26 + ord('a'))
        else:
            ciphertext += p

    print("ciphertext:", ciphertext)


if __name__ == "__main__":
    main()
